core/src/main/scala/org/http4s/util/UrlCoding.scala

```diff
@@ -91,7 +117,10 @@ private[util] trait UrlCodingUtils {
             in.position(mark+1)
           }
         } else {
-          while(in.hasRemaining) in.get() // just burn the rest of the bytes
+          // This is an invalid encoding. Fail gracefully by treating the '%' as
```

Comment:
I don't think there is a graceful thing to do here since its invalid, but either way is fine with me. :)